server/channels/app/post.go

```diff
@@ -461,6 +461,7 @@ func (a *App) attachFilesToPost(post *model.Post) *model.AppError {
 // If channel is nil, FillInPostProps will look up the channel corresponding to the post.
 func (a *App) FillInPostProps(c request.CTX, post *model.Post, channel *model.Channel) *model.AppError {
 	channelMentions := post.ChannelMentions()
+	ChannelMentionsAcrossTeams := post.ChannelMentionsAcrossTeams()
```

Comment:
Let's also add a test similar to `TestPostChannelMentions` to test this functionality and verify that the post prop gets set properly.